app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,9 @@
+class FoodsController < ApplicationController
+  def index
+    @searched_ingredient = params[:q]
+    
+    @total_results = FoodFacade.total_results_ingredient(@searched_ingredient)
+
+    @foods = FoodFacade.foods_with_ingredient(@searched_ingredient)
+  end
```

Comment:
You're code's looking good.. so... a little challenge for ya 😏 : 

Both of these methods, `total_results_ingredient` and `foods_with_ingredient` call the same endpoint, and just pull different data. I'd like to challenge you to find a way to just call one method in your facade, and have your facade take care of combining the food objects and the totalHits into one object to get returned to your controller. All of the data that you need for this view can come from the one query call you're making.

Reply:
You could take a look at this syntax:

@total_results, @foods = FoodFacade.your_method(params[:q])

Where your_method returns a 2d array, with the number of results as the first element and the food collection as the second element.